src/widgets/breadcrumb/breadcrumb.tsx

```diff
@@ -42,97 +45,88 @@ const renderer = ({ containerNode, cssClasses, renderState, templates }) => (
- * It is commonly used for websites with a large amount of data organized into categories with subcategories.
- *
- * All attributes (lvl0, lvl1 in this case) must be declared as [attributes for faceting](https://www.algolia.com/doc/guides/searching/faceting/#declaring-attributes-for-faceting) in your
- * Algolia settings.
- *
- * @requirements
- * Your objects must be formatted in a specific way to be
- * able to display a breadcrumb. Here's an example:
- *
- * ```javascript
- * {
- *   "objectID": "123",
- *   "name": "orange",
- *   "categories": {
- *     "lvl0": "fruits",
- *     "lvl1": "fruits > citrus"
- *   }
- * }
- * ```
- *
- * Each level must be specified entirely.
- * It's also possible to have multiple values per level, for instance:
- *
- * ```javascript
- * {
- *   "objectID": "123",
- *   "name": "orange",
- *   "categories": {
- *     "lvl0": ["fruits", "vitamins"],
- *     "lvl1": ["fruits > citrus", "vitamins > C"]
- *   }
- * }
- * ```
- * @type {WidgetFactory}
- * @devNovel Breadcrumb
- * @category navigation
- * @param {BreadcrumbWidgetOptions} $0 The Breadcrumb widget options.
- * @return {Widget} A new Breadcrumb widget instance.
- * @example
- * search.addWidgets([
- *   instantsearch.widgets.breadcrumb({
- *     container: '#breadcrumb',
- *     attributes: ['hierarchicalCategories.lvl0', 'hierarchicalCategories.lvl1', 'hierarchicalCategories.lvl2'],
- *     templates: { home: 'Home Page' },
- *     separator: ' / ',
- *     rootPath: 'Cameras & Camcorders > Digital Cameras',
- *   })
- * ]);
- */
-
-export default function breadcrumb({
-  container,
-  attributes,
-  separator,
-  rootPath = null,
-  transformItems,
-  templates = defaultTemplates,
-  cssClasses: userCssClasses = {},
-} = {}) {
+export type BreadcrumbCSSClasses = {
+  /**
+   * CSS class to add to the root element of the widget.
+   */
+  root?: string | string[];
+
+  /**
+   * CSS class to add to the root element of the widget if there are no refinements.
+   */
+  noRefinementRoot?: string | string[];
+
+  /**
+   * CSS class to add to the list element.
+   */
+  list?: string | string[];
+
+  /**
+   * CSS class to add to the items of the list. The items contains the link and the separator.
+   */
+  item?: string | string[];
+
+  /**
+   * CSS class to add to the selected item in the list: the last one or the home if there are no refinements.
+   */
+  selectedItem?: string | string[];
+
+  /**
+   * CSS class to add to the separator.
+   */
+  separator?: string | string[];
+
+  /**
+   * CSS class to add to the links in the items.
+   */
+  link?: string | string[];
+};
+
+export type BreadcrumbTemplates = {
+  /**
+   * Label of the breadcrumb's first element.
+   */
+  home?: string | ((value: any) => string);
+
+  /**
+   * Symbol used to separate the elements of the breadcrumb.
+   */
+  separator?: string | ((value: any) => string);
```

Comment:
Can you confirm that this also accepts a function?

Reply:
you can use the `Template` or `TemplateData` type we have. Since no value is given to those you can use a string too (for any template), but there won't be any value given, so it should be `() => string`